packages/@react-aria/color/src/useColorSlider.ts

```diff
@@ -61,6 +62,43 @@ export function useColorSlider(props: ColorSliderAriaOptions, state: ColorSlider
     inputRef
   }, state);
 
+  let {minValue, maxValue, step, pageSize} = state.value.getChannelRange(channel);
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      // these are the cases that useMove or useSlider don't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, stopPropagation to prevent useSlider from handling the event as well.
+      e.preventDefault();
+      let channelValue = state.value.getChannelValue(channel);
+      let pageStep = Math.max(pageSize, step);
+      let newValue = channelValue;
+      switch (e.key) {
+        case 'PageUp':
+          newValue = channelValue + pageStep > maxValue ? maxValue : snapValueToStep(channelValue + pageStep, minValue, maxValue, pageStep);
+          break;
+        case 'PageDown':
+          newValue = snapValueToStep(channelValue - pageStep, minValue, maxValue, pageStep);
+          break;
+        case 'Home':
+          newValue = minValue;
+          break;
+        case 'End':
+          newValue = maxValue;
+          break;
+      }
+      if (newValue !== channelValue) {
+        // remember to set this so that onChangeEnd is fired
+        state.setThumbDragging(0, true);
+        state.setValue(state.value.withChannelValue(channel, newValue));
+        // wait a frame to ensure value has changed then unset this so that onChangeEnd is fired
+        requestAnimationFrame(() => state.setThumbDragging(0, false));
```

Comment:
Do we need this raf? I didn't see one in the other hooks, what make this a special case?

Reply:
Yes, I do need to wait a frame, otherwise the value returned by `onChangeEnd` will be the value that preceded that returned by the `onChange` event.